hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/PeekWrappedP.java

```diff
@@ -242,21 +242,18 @@ public boolean offer(int ordinal, @Nonnull Object item) {
         public boolean offer(int[] ordinals, @Nonnull Object item) {
             // use broadcast logic to be able to report accurately
             // which queue was pushed to when.
-            boolean done = true;
             for (int i = 0; i < ordinals.length; i++) {
                 if (broadcastTracker.get(i)) {
                     continue;
                 }
                 if (offer(i, item)) {
                     broadcastTracker.set(i);
                 } else {
-                    done = false;
+                    return false;
```

Comment:
slight difference here, it will stop at first full queue rather than trying to push everywhere and then checking the status. Is it intentional?

Reply:
yes because after first full queue `false` is returned and we need to call `reset()`.